timeskew/timeskew.go

```diff
@@ -0,0 +1,117 @@
+package timeskew
+
+import (
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/beevik/ntp"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	// DefaultServer will be internally resolved to the closest available.
+	// also it rarely queries same server more than once.
+	DefaultServer = "pool.ntp.org"
+
+	// DefaultAttempts defines how many servers we will query
+	DefaultAttempts = 3
+
+	// DefaultUpdatePeriod defines how often time will be queried from ntp.
+	DefaultUpdatePeriod = 2 * time.Minute
+)
+
+type ntpQuery func(string) (*ntp.Response, error)
```

Comment:
You're interested in the offset, so maybe return the duration and use a wrapper to hide NTP or any other way to get this offset.